File: generator/src/test/resources/swift/expected/RequestMixedParamsTest/test-generation-of-multiple-parameters-with-inline-type-definitions.output.swift
import PotentCodables
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest(
    select: FetchTestSelectUriParam,
    page: FetchTestPageQueryParam,
    xType: FetchTestXTypeHeaderParam
  ) throws -> Sunday.Operation<Empty, [String : AnyValue], TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests/{select}",
        pathParameters: [
          "select": try ParameterValues.encode(select)
        ],
        queryParameters: [
          "page": try ParameterValues.encode(page)
        ],
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: [
          "x-type": try ParameterValues.encode(xType)
        ]
      )
    )
  }

  public enum FetchTestSelectUriParam : String, CaseIterable, Codable, Sendable {

    case all = "all"
    case limited = "limited"

  }

  public enum FetchTestPageQueryParam : String, CaseIterable, Codable, Sendable {

    case all = "all"
    case limited = "limited"

  }

  public enum FetchTestXTypeHeaderParam : String, CaseIterable, Codable, Sendable {

    case all = "all"
    case limited = "limited"

  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestMixedParamsTest/test-generation-of-multiple-parameters-with-inline-type-definitions.output.kt
package io.test.service

import com.fasterxml.jackson.`annotation`.JsonCreator
import com.fasterxml.jackson.`annotation`.JsonValue
import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.Operation
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.operation
import java.lang.IllegalArgumentException
import kotlin.Any
import kotlin.String
import kotlin.Unit
import kotlin.collections.List
import kotlin.collections.Map
import kotlin.jvm.JvmStatic

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public fun fetchTest(
    select: FetchTestSelectUriParam,
    page: FetchTestPageQueryParam,
    xType: FetchTestXTypeHeaderParam,
  ): Operation<Unit, Map<String, Any>, Req> = this.transport.operation<Unit, Map<String, Any>, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/tests/{select}",
      pathParameters = mapOf(
        "select" to select
      ),
      queryParameters = mapOf(
        "page" to page
      ),
      acceptTypes = this.defaultAcceptTypes,
      headers = mapOf(
        "x-type" to xType
      )
    )
  )

  public enum class FetchTestSelectUriParam(
    private val wireValue: String,
  ) {
    All("all"),
    Limited("limited"),
    ;

    @JsonValue
    public override fun toString(): String = wireValue

    public companion object {
      @JsonCreator
      @JvmStatic
      public fun fromValue(rawValue: String): FetchTestSelectUriParam {
        for (entry in entries) {
          if (entry.wireValue == rawValue) {
            return entry
          }
        }
        throw IllegalArgumentException("Unknown FetchTestSelectUriParam value: " + rawValue)
      }
    }
  }

  public enum class FetchTestPageQueryParam(
    private val wireValue: String,
  ) {
    All("all"),
    Limited("limited"),
    ;

    @JsonValue
    public override fun toString(): String = wireValue

    public companion object {
      @JsonCreator
      @JvmStatic
      public fun fromValue(rawValue: String): FetchTestPageQueryParam {
        for (entry in entries) {
          if (entry.wireValue == rawValue) {
            return entry
          }
        }
        throw IllegalArgumentException("Unknown FetchTestPageQueryParam value: " + rawValue)
      }
    }
  }

  public enum class FetchTestXTypeHeaderParam(
    private val wireValue: String,
  ) {
    All("all"),
    Limited("limited"),
    ;

    @JsonValue
    public override fun toString(): String = wireValue

    public companion object {
      @JsonCreator
      @JvmStatic
      public fun fromValue(rawValue: String): FetchTestXTypeHeaderParam {
        for (entry in entries) {
          if (entry.wireValue == rawValue) {
            return entry
          }
        }
        throw IllegalArgumentException("Unknown FetchTestXTypeHeaderParam value: " + rawValue)
      }
    }
  }
}
